Automattic/jetpack
# Subscribe block: don't include social followers on subscriber counts by default

Update "include social follows in count" to be disabled by default.

## Proposed changes:

* Don't include social followers on subscriber counts by default
* Don't hide the social count toggle, simply disable it. Makes for less jumpy UI.

**Before**, enabling social count would set "include social counts" active by default as well

<img width="1318" alt="Screenshot 2023-12-13 at 14 39 20" src="https://github.com/Automattic/jetpack/assets/87168/7b188238-d740-4fc8-945f-e90f4fd7b4da">

**After**, "include social counts" stays disabled until enabled specifically:

<img width="1061" alt="Screenshot 2023-12-13 at 14 47 13" src="https://github.com/Automattic/jetpack/assets/87168/166bd1eb-cc9e-46ad-b365-857104eaa269">

<img width="1054" alt="Screenshot 2023-12-13 at 14 47 20" src="https://github.com/Automattic/jetpack/assets/87168/e620d7ba-9392-46b0-a27c-8cce62460fb1">

We cannot just switch "default" in the `block.json` to false, because block editor doesn't store values matching the default in block's saved attributes. Hence every block with "include social counts" enabled (which was the default before) _does not_ have this value stored. Changing the default to `false` would switch all those blocks to not include social counts, hence altering their numbers and sites.

Hence we manually check for `undefined` when toggle is touched, and always store both `true`/`false` values for `includeSocialFollowers` in the attributes now. 

Altering the numbers (albeit somewhat vanity number) would be controbersial.

Note that since Twitter connections have been disconnected for a while, we haven't been able to update the social counts for sites anyway. They're stuck in the past counts.

### Other information:



## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Insert block, go to settings and enable the "show counts".
* Ensure social counts toggle stays disabled, and value is stored in attributes when looking at block code (three dots menu from the top):
    <img width="811" alt="Screenshot 2023-12-13 at 15 03 24" src="https://github.com/Automattic/jetpack/assets/87168/78ba0046-d3f1-43c5-838c-37f4abf62885">
* Enabling the "include social counts" includes in the code now as well:
   <img width="816" alt="Screenshot 2023-12-13 at 15 03 44" src="https://github.com/Automattic/jetpack/assets/87168/8b79a6bf-3326-452c-ac0d-a1dd48e325d1">
* Manually remove the `includeSocialFollowers` attribute from block markup and save the post.
* When previewing the block on the frontend, log `$include_social_followers` value and see it's `true`.
* Open the block with missing `` atrribute in the editor, and confirm that toggle appears "enabled" (since unset value defaults to enabled). Setting the value on/off still works.
    <img width="715" alt="Screenshot 2023-12-13 at 15 15 45" src="https://github.com/Automattic/jetpack/assets/87168/6bfc67e6-7052-4990-a218-39a88bf913fe">
    <img width="306" alt="Screenshot 2023-12-13 at 15 16 14" src="https://github.com/Automattic/jetpack/assets/87168/07746029-05a5-4952-ad82-ca08638dd7bf">
